### ISSUE TITLE
"Client ID is already in use" causing non-recoverable disconnects

### ISSUE THREAD

**Issue by Maharaghi** [cloudradar-monitoring/rport#566]
Hello!
I started deploying rport for a number of clients a few weeks ago, and it's lovely.
But I've encountered, I think, a very crucial bug. 

#### TL;DR of what I'm pretty sure the bug is.
The check on [this line](https://github.com/cloudradar-monitoring/rport/blob/fc1dd8f60a0a54a12d07ea8416c07025f76d8120/client/client.go#L353) does not match with the response the [server sends](https://github.com/cloudradar-monitoring/rport/blob/fc1dd8f60a0a54a12d07ea8416c07025f76d8120/server/client_service.go#L225).  
This causes the client to exit and not try to recover, which leads to a full disconnect and you need to manually restart the service to fix it.  
A solution seems to be to simply use `if strings.Contains(msg, "is already in use")`, or something similar on the client side (maybe make it identical to the server with the client id as well).  

#### Boring stuff about how I noticed
I noticed my work-pc constantly disconnecting from the rport server, and never reconnecting. So I enabled debug logs to see where it stopped working. After sending the connection request it just completely stopped.  

This happened a lot for my pc because I switch between ~10 vpn connections to manage different servers (hence why I want rport) and the client would then sometimes briefly disconnect during the VPN connection, causing this error when reconnecting.  
I did some digging and found the server+client "id in use" string, changed the client one to `if strings.Contains(msg, "is already in use")` and that seems to have done the trick.

I hope this bug is simply that single string!

**P.S**
The client doesn't seem to be [logging any fatal errors ](https://github.com/cloudradar-monitoring/rport/blob/fc1dd8f60a0a54a12d07ea8416c07025f76d8120/client/client.go#L200)returned from `sendConnectionRequest`, which made me take a bit longer to figure out the problem. Perhaps this should be added as well?

**Comment by thorstenkramm** [comment]
@Maharaghi Wow! This is an outstanding bug report. Thank you so much for finding that out. 🙏 
We will shortly integrate the needed changes into the code base.
